eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/steps/tasklets/reportCheck/AccessionWrapperComparator.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.pipeline.steps.tasklets.reportCheck;
+
+import uk.ac.ebi.ampt2d.commons.accession.core.AccessionWrapper;
+
+import uk.ac.ebi.eva.accession.core.ISubmittedVariant;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class AccessionWrapperComparator implements Comparator<AccessionWrapper<ISubmittedVariant, String, Long>> {
+
+    private Map<String, Integer> contigOrder;
+
+    public AccessionWrapperComparator(List<? extends ISubmittedVariant> variants) {
+        this.contigOrder = getContigOrder(variants);
+    }
+
+    private Map<String, Integer> getContigOrder(List<? extends ISubmittedVariant> variants) {
```

Comment:
I think that iterating all the variants just to get the contig order is a little bit too much. In the other hand, they will be the  variants in a single Batch that is already in memory, so it shouldn't take long and the impact of this will be minimal. And this is being made to improve the performance of the checking algorithm, so probably the penalty is paid off later.

If we see this validation is too slow when we start testing with real data, we can take a look at this, otherwise, is OK like it is.